docker/entrypoint.sh

```diff
@@ -1,12 +1,112 @@
 #!/bin/bash
+
 set -e
+set -o pipefail
+
+PGDATA=${PGDATA:=/var/lib/postgresql/9.6/main}
+PGUSER=${PGUSER:=postgres}
+PGPASSWORD=${PGPASSWORD:=postgres}
+PG_STARTUP_WAIT_TIME=${PG_STARTUP_WAIT_TIME:=10}
+SUPERVISORD_LOG_LEVEL=${SUPERVISORD_LOG_LEVEL:=info}
+PGBIN="/usr/lib/postgresql/9.6/bin"
+
+if [ ! -z "$ATTACHMENTS_STORAGE_PATH" ]; then
+	mkdir -p "$ATTACHMENTS_STORAGE_PATH"
+	chown -R app:app "$ATTACHMENTS_STORAGE_PATH"
+fi
 
 if [ "$(id -u)" = '0' ]; then
-	chown -R $APP_USER:$APP_USER $APP_PATH $APP_DATA /usr/local
-	sync
-	exec $APP_PATH/docker/gosu $APP_USER "$BASH_SOURCE" "$@"
+    echo "-----> Ensure $APP_PATH is owned by $APP_USER"
+	mkdir -p "$APP_PATH/log" "$APP_PATH/tmp" "$APP_PATH/files"
+	chown $APP_USER:$APP_USER "$APP_PATH"
+	chown -R $APP_USER:$APP_USER "$APP_PATH/log" "$APP_PATH/tmp" "$APP_PATH/files" "$APP_PATH/public"
+fi
+
+dbhost=$(ruby -ruri -e 'puts URI(ENV.fetch("DATABASE_URL")).host')
+pwfile=$(mktemp)
+echo "$PGPASSWORD" > $pwfile
+chown postgres $pwfile
+
+PLUGIN_GEMFILE_TMP=$(mktemp)
+PLUGIN_GEMFILE=/usr/src/app/Gemfile.local
+
+if [ "$PLUGIN_GEMFILE_URL" != "" ]; then
+	echo "Fetching custom gemfile from ${PLUGIN_GEMFILE_URL}..."
+	curl -L -o "$PLUGIN_GEMFILE_TMP" "$PLUGIN_GEMFILE_URL"
+
+	# set custom plugin gemfile if file is readable and non-empty
+	if [ -s "$PLUGIN_GEMFILE_TMP" ]; then
+		mv "$PLUGIN_GEMFILE_TMP" "$PLUGIN_GEMFILE"
+		chown app.app "$PLUGIN_GEMFILE"
+	fi
+fi
+
+indent() {
+	sed -u 's/^/       /'
+}
+
+install_plugins() {
+	pushd /usr/src/app
+
+	if [ -s "$PLUGIN_GEMFILE" ]; then
+		echo "Installing plugins..."
+		bundle install
+		echo "Precompiling new assets..."
+		bundle exec rake assets:precompile
```

Comment:
@oliverguenther This won't work with the deleted frontend / npm modules folder, will it?

Reply:
Good point, didn't know this was even supported. It should suffice to run `npm install` in this step.